tools/CopyrightDateTool.scala

```diff
@@ -0,0 +1,167 @@
+    val res = gits_info.filter(!_.message.contains("[Sync copyright dates]"))
+    val year_column = 4
+    val first = 0
+    res(first).date.split(" ")(year_column)
+  }
+
+  def getYear(pathToFile: String, lower_bound: Int, upper_bound: Int): String = {
+    val file = readFile(pathToFile)
+    val third_line = 2
+    val start_index = file.split("\\n")(third_line).indexOf("(c) 20")
+    val year = file.split("\\n")(third_line)
+                   .substring(start_index + lower_bound,
+                              start_index + upper_bound)
+    year
+  }
+
+  def getFileCreationYear(pathToFile: String): String = {
+    val lb = "(c) ".length()
+    val ub = "(c) 20XX".length()
+    val fileCreationYear = getYear(pathToFile, lb, ub)
+    fileCreationYear
+  }
+
+  def getFileLastUpdateYear(pathToFile: String): String = {
+    val lb = "(c) 20XX-".length()
+    val ub = "(c) 20XX-XXXX".length()
+    val fileLastUpdateYear = getYear(pathToFile, lb, ub)
+    fileLastUpdateYear
+  }
+
+  def analyzeDates(): Array[FileDates] = {
+    val files_array = readFilesList()
+    val data = files_array map {
+      case file_path =>
+      val fileLastUpdateYear = getFileLastUpdateYear(file_path)
+      val gitLastUpdateYear = getGitLastUpdateYear(file_path)
+      val fileCreationYear = getFileCreationYear(file_path)
+      val gitCreationYear = getGitCreationYear(file_path)
+      FileDates(file_path,
+                fileLastUpdateYear,
+                gitLastUpdateYear,
+                fileCreationYear,
+                gitCreationYear)
+     }
+    data
+  }
+
+  def fixWrongDates(): Boolean = {
+    var check = false
+    val files = analyzeDates()
+    files.foreach { file =>
+      val file_file = readFile(file.pathToFile)
+      var new_file: Option[String] = None
+      if (file.fileLastUpdateYear.forall(_.isDigit)) {
+        if (file.isUpdateDateWrong()) {
+          new_file =Some(file_file.
+                       replace(
+                        s"${file.fileCreationYear}-${file.fileLastUpdateYear}",
+                        s"${file.fileCreationYear}-${file.gitLastUpdateYear}"
+                       ))
+        }
+      }
+      if(new_file.isDefined) {
+        scala.tools.nsc.io.File(file.pathToFile)("UTF-8").writeAll(new_file.get)
+        check = true 
+      }
+    }
+    check
+  }
+
+  def resultAnalyzeDates(): Boolean = {
+    val data = analyzeDates
+    val (error_files, ok_files) = data.partition(x =>
+      x.isUpdateDateWrongStr() &&
+      x.fileLastUpdateYear.forall(_.isDigit))
+    if (error_files.nonEmpty)
+    {
+      println("""
+        Format
+        FileDates(file_path,file update year,git update year, file creation year, git creation year)
+        """)
+      error_files.foreach { file => println(file) }
+      println("Oops...We have some error dates")
+    }
+    else println ("Everything is ok")
+    error_files.nonEmpty
+  }
+
+  def main(args: Array[String]) {
+    if (args.length != 1) println(usage)
```

Comment:
If this line was removed it would prevent the usage message from being printed twice when the number of args is <> 1.